### ISSUE TITLE
There's a resource leak somewhere in the App/Control/Connection stack

### ISSUE THREAD

**Issue by Kirbk** [microsoft/terminal#745]
When closing a tab, the memory that was allocated for the tab is retained. This leads to a massive memory leak when the application is used continuously, opening and closing tabs like users do. This uses gigabytes of RAM, the only way to clear it is to close out of the terminal completely and relaunch it.

**Comment by onekopaka** [comment]
I do not have very concrete data on it yet, but I also see a massive memory leak in WindowsTerminal.exe, even with only ever having just one tab open.

I tend to see this memory consumption happen most aggressively with any sort of application that continuously updates the screen like mtr or htop (I'm using the version of OpenSSH bundled with Windows build 18362 to connect to a Linux machine to run these). A session that run for just under 40 minutes got up to over 1000 MB of memory usage according to Task Manager.

**Comment by onekopaka** [comment]
Another bit of information, all that memory seems to be released over time, that same session, now sitting at a shell over SSH calms down and now roughly 3 hours later, it is using an efficient and trim 8.5 MB of RAM according to Task Manager.

**Comment by zadjii-msft** [comment]
@onekopaka I moved your bug report to 768, to try and keep this thread specific to the cleanup of the TermControl's when a tab is closed

**Comment by dreamer-dead** [comment]
Is it possible to reproduce this leak after #914 was merged into `master`?
I can't check that by myself =(

**Comment by Kirbk** [comment]
@dreamer-dead I just rechecked and it still appears to still be present in the most recent merge.

**Comment by driver1998** [comment]
I guess when the tab closes, it doesn't end the conhost process attached to it.

What I did:
Start Windows Terminal, open 9 powershell tabs, and close 8 of them.
Result:
![TIM截图20190530123246](https://user-images.githubusercontent.com/22699485/58608616-11386980-82d7-11e9-87fd-11e24c5150d6.png)

**Comment by zadjii-msft** [comment]
Somehow, I fixed this in #825. Hooray!
